src/gluonts/dataset/parallelized_loader.py

```diff
@@ -0,0 +1,665 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+
+# Standard library imports
+import itertools
+import pickle
+import io
+import random
+import sys
+import time
+from typing import Callable, Iterable
+
+import multiprocessing
+import multiprocessing.queues
+from multiprocessing.reduction import ForkingPickler
+from multiprocessing.pool import Pool
+from multiprocessing import Queue
+
+try:
+    import multiprocessing.resource_sharer
```

Comment:
When does this happen?

Reply:
This is copied from MXNet, so I'm not exactly sure why they decided to do it this way.